Frends.Web/Web.cs

```diff
@@ -287,11 +287,22 @@ public class Web
             httpClient.DefaultRequestHeaders.ExpectContinue = false;
             httpClient.Timeout = TimeSpan.FromSeconds(Convert.ToDouble(options.ConnectionTimeoutSeconds));
 
-            using (HttpContent content = new StringContent(input.Message ?? ""))
+            //Ignore case for headers and key comparison
+            var headerDict = input.Headers.ToDictionary(key => key.Name, value => value.Value, StringComparer.InvariantCultureIgnoreCase);
+
+            //Check if Content-Type exists and is set and valid
+            var contentTypeIsSetAndValid = false;
+            MediaTypeWithQualityHeaderValue validContentType = null;
+            if (headerDict.TryGetValue("content-type", out string contentTypeValue))
+                contentTypeIsSetAndValid = MediaTypeWithQualityHeaderValue.TryParse(contentTypeValue, out validContentType);
+
+            using (HttpContent content = contentTypeIsSetAndValid ?
+                new StringContent(input.Message ?? "", Encoding.GetEncoding(validContentType.CharSet)) :
```

Comment:
What happens here if the contentType is for example application/json and no charset is given? Will the charset get a default value of utf-8 or will it be null? GetEncoding will probably not work with a null value.

Reply:
You're right, apparently TryParse returns true without charset.
Defaulting to utf-8 webname if null is a fix. It would allow user to set Content-Type without charset parameter.